frigate/detectors/detection_api.py

```diff
@@ -8,10 +10,23 @@
 
 
 class DetectionApi(ABC):
+    _api_types = {}
+
     @abstractmethod
     def __init__(self, det_device=None, model_config=None):
         pass
 
     @abstractmethod
     def detect_raw(self, tensor_input):
         pass
+
+    @staticmethod
+    def register_api(det_type: DetectorTypeEnum, det_api):
+        DetectionApi._api_types[det_type] = det_api
+
+    @staticmethod
+    def create(det_type: DetectorTypeEnum, **kwargs):
```

Comment:
Should this create function be separate from the DetectionApi class? This is the parent class of the detectors, but it doesn't make sense for each them to have this as a method?

Reply:
If this function were defined in the __init__.py file and had access to a static map of the enum to api, I think we could also eliminate the need to have the `register_api` method?

Reply:
I slept on it, and I think you may be right.  I think I like this register/create mechanism as-is.

Reply:
this can probably be marked as resolved then?

Reply:
If I was smart and made these a review instead of comments I could?  I don't get a mark resolved button here though. Lesson learned for next time.

Reply:
Weird, I have one so I'll mark it as resolved.

Reply:
oops, I missed the last couple of comments here and made the change already! sorry! I was going back and forth on whether to defend the register/create mechanism as well, so thats why I made the change but I'm going to revert this back.

Reply:
@dennispg I wouldn't revert it, I think I like the way this is now TBH. Curious what @NateMeyer thinks

Reply:
Yea, I'll wait for your feedback before I revert. In the meantime, I'm going to take a stab at the model configs

Reply:
I think my initial concern was because I was misunderstanding the `@staticmethod` decorator.  I'd be curious to hear your reasons for doing it that way initially.

Reply:
It felt like a concern that is related to the DetectionApi type, but probably the biggest reason is because I'm more of a C# dev than python...

I think I agree with @NickM-27 now though, it feels more "pythonic" now.